content/en/docs/sdk-utilities/s3.md

```diff
@@ -101,6 +101,48 @@ before sending it to {{% alias service="S3" %}}. `Uploader` calculates the expec
+
+```go
+result, err := uploader.Upload(context.TODO(), &s3.PutObjectInput{
+	Bucket: aws.String("my-bucket"),
+	Key:    aws.String("my-object-key"),
+	Body:   uploadFile,
+	ChecksumAlgorithm: types.ChecksumAlgorithmSha256,
+})
+```
+
+If an explicit checksum value in either the `ChecksumCRC32`, `ChecksumCRC32C`, `ChecksumSHA1`, or `ChecksumSHA256` fields is specified, then the SDK will send that value specified using a standard HTTP request header using the appropriate `x-amz-checksum-*` header for each. This is not affected by whether the HTTP or HTTPS scheme is used. The request will be signed with SigV4 or SigV4a regardless of the scheme, the only distinction is whether the payload body hash is computed or not for the `x-amz-content-sha256` header. For HTTP schemes this will be the computed sha256 hash for the `PutObject` body, otherwise for HTTPS this will be `UNSIGNED-PAYLOAD`. If the payload body hash needs to be computed for sending to a non-HTTPS endpoint, then the body must be seekable so the SDK can compute the hash and reseek the body.
+
+If `ChecksumAlgorithm` field, and an algorithm `Checksum*` member is not set, then the SDK will compute the corresponding checksum for the request and will send it in the appropriate `x-amz-checksum-*` header in the REST request. The only distinction is if the checksum is sent as a header or trailer based on the endpoint scheme.
+
+* If the scheme is HTTPS the computed checksum will be sent in a trailer header block using aws-chunked encoding. It will also do so use an unsigned payload variant called `STREAMING-UNSIGNED-PAYLOAD-TRAILER` which does not require signing each aws-chunk. The Go SDK only supports this variant of aws-chunked, which is why computed checksums sent via a aws-chunked trailer requires an HTTPS endpoint in this SDK. For trailer streaming the content-length of the `PutObject` body must be known. So either the `PutObjectInput` specifies `ContentLength` field, the `Body` field implements `interface{ Len() int }`, or the body is seekable so SDK can figure out the length to be sent. The SDK may add support for the signed aws-chunked variants (e.g. `STREAMING-AWS4-HMAC-SHA256-PAYLOAD-TRAILER`) in the future as to support sending of signed checksum trailers to non-HTTPS endpoints. That feature is supported in SDKs that had already supported signed aws-chunked payloads previously. For more information about checksum location and trailing checksum support for unsigned payload, please refer to [flexible-checksums](https://code.amazon.com/packages/AwsDrSeps/blobs/main/--/seps/accepted/shared/flexible-checksums.md) 
```

Comment:
this isn't true, where are you getting this information?

Reply:
Ok we can talk about that, I'm not 100% sure that is all accurate but it's also talking about S3 integrity features for `PutObject`/`UploadPart` operations. We can document those in a different section and link to them from here. This section is on the S3 transfer manager which I think we should keep the discussion here related to the differences.